Packages/com.unity.inputsystem/InputSystem/Editor/AssetEditor/InputActionEditorToolbar.cs

```diff
@@ -0,0 +1,661 @@
+#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.IMGUI.Controls;
+using UnityEditorInternal;
+using UnityEngine.Experimental.Input.Utilities;
+
+////TODO: better method for creating display names than InputControlPath.TryGetDeviceLayout
+
+namespace UnityEngine.Experimental.Input.Editor
+{
+    /// <summary>
+    /// Toolbar in input action asset editor.
+    /// </summary>
+    /// <remarks>
+    /// Allows editing and selecting from the set of control schemes as well as selecting from the
+    /// set of device requirements within the currently selected control scheme.
+    ///
+    /// Also controls saving and has the global search text field.
+    /// </remarks>
+    /// <seealso cref="InputActionEditorWindow"/>
+    [Serializable]
+    internal class InputActionEditorToolbar
+    {
+        public void OnGUI()
+        {
+            EditorGUILayout.BeginHorizontal(EditorStyles.toolbar);
+            DrawSchemeSelection();
+            DrawDeviceFilterSelection();
+            if (!InputEditorUserSettings.autoSaveInputActionAssets)
+                DrawSaveButton();
+            GUILayout.FlexibleSpace();
+            DrawAutoSaveToggle();
+            GUILayout.Space(5);
+            DrawSearchField();
+            GUILayout.Space(5);
+            EditorGUILayout.EndHorizontal();
+        }
+
+        private void DrawSchemeSelection()
+        {
+            var buttonGUI = new GUIContent(selectedControlScheme?.name ?? "No Control Scheme");
```

Comment:
Good point. Changed it so that

* If you have no control schemes in the asset, it shows "No Control Schemes" and the popup also doesn't list any (previously it presented "No Control Scheme" like a control scheme).
* If you do have control schemes, then what used to be "No Control Scheme" is now "All Control Schemes" (including "Show " thing is a bit long).

Works?

Reply:
Works, but: 
>"All Control Schemes" (including "Show " thing is a bit long).

I included the word "Show" there on purpose. Because when I first tried this menu, it was not clear to me that it acted as a "filter" controlling which parts of the data I see (as opposed to, eg, performing some reformatting of the data to make it suitable for different things, which could be an action actually changing the data). I was thinking about making a change to communicate that better. I considered suggesting a "Filter: " prefix label for this menu, but I figured that that might be unnecessary UI clutter, so I thought that adding "Show" to the label might be a lightweight but sufficient clue to the user.